xilem/examples/stopwatch.rs

```diff
@@ -0,0 +1,263 @@
+    let hours = (dur.as_secs() / 60) / 60;
+    format!("{hours}:{minutes:0>2}:{seconds:0>4.1}")
+}
+
+fn app_logic(data: &mut Stopwatch) -> impl WidgetView<Stopwatch> {
+    fork(
+        flex((
+            FlexSpacer::Fixed(5.0),
+            label(get_formatted_duration(&data.displayed_duration)).text_size(70.0),
+            flex((lap_reset_button(data), start_stop_button(data))).direction(Axis::Horizontal),
+            FlexSpacer::Fixed(1.0),
+            laps_section(data),
+            label(data.displayed_error.as_ref()),
+        ))
+        .main_axis_alignment(MainAxisAlignment::Start)
+        .cross_axis_alignment(CrossAxisAlignment::Center),
+        async_repeat(
+            |proxy| async move {
+                let mut interval = time::interval(Duration::from_millis(50));
+                loop {
+                    interval.tick().await;
+                    let Ok(()) = proxy.message(()) else {
+                        break;
+                    };
+                }
+            },
+            |data: &mut Stopwatch, ()| {
+                if data.active {
+                    data.update_display();
+                }
+            },
+        ),
+    )
+}
+
+/// Creates a list of items that shows the lap number, split time, and total cumulative time.
+fn laps_section(data: &mut Stopwatch) -> Vec<AnyFlexChild<Stopwatch>> {
+    let mut items: Vec<AnyFlexChild<Stopwatch>> = Vec::new();
+    let mut total_dur = Duration::ZERO;
+    let current_lap = data.completed_lap_splits.len();
+
+    for (i, split_dur) in data.completed_lap_splits.iter().enumerate() {
+        total_dur = total_dur.add(*split_dur);
+        items.push(single_lap(i, split_dur, &total_dur).into_any_flex());
+    }
+    let current_split_duration = data.get_current_duration().sub(total_dur);
+    let mut reversed = Vec::new();
+    reversed.push(
+        single_lap(
+            current_lap,
+            &current_split_duration,
+            &data.get_current_duration(),
+        )
+        .into_any_flex(),
+    );
+    for item in items.into_iter().rev() {
+        reversed.push(item);
+    }
+    reversed
+}
+
+fn single_lap(
+    lap_id: usize,
+    split_dur: &Duration,
+    total_dur: &Duration,
+) -> impl WidgetView<Stopwatch> {
+    flex((
+        FlexSpacer::Flex(1.0),
+        label(format!("Lap {}", lap_id + 1)).flex(1.0),
+        label(get_formatted_duration(split_dur)).flex(1.0),
+        label(get_formatted_duration(total_dur)).flex(1.0),
+    ))
+    .direction(Axis::Horizontal)
+    .cross_axis_alignment(CrossAxisAlignment::Center)
+    .main_axis_alignment(MainAxisAlignment::Center)
+}
+
+fn start_stop_button(data: &mut Stopwatch) -> AnyFlexChild<Stopwatch> {
+    if data.active {
+        button("Stop", |data: &mut Stopwatch| {
+            data.stop();
+        })
+        .into_any_flex()
+    } else {
+        button("Start", |data: &mut Stopwatch| {
+            data.start();
+        })
+        .into_any_flex()
+    }
+}
```

Comment:
```suggestion
fn start_stop_button(data: &mut Stopwatch) -> impl WidgetView<Stopwatch> {
    if data.active {
        Either::A(button("Stop", |data: &mut Stopwatch| {
            data.stop();
        }))
    } else {
        Either::B(button("Start", |data: &mut Stopwatch| {
            data.start();
        }))
    }
}
```

Reply:
That's what I needed. Thank you. This example will be a good example for future users of Xilem.
I did this for both buttons.